tests/mocha/procedure_map_test.js

```diff
@@ -200,8 +201,494 @@ suite('Procedure Map', function() {
+              parameter: parameterModel,
+              index: 0,
+            },
+            this.workspace.id);
+        });
+
+    test(
+        'parameter create events are not fired if the procedure is ' +
+        'not in the map',
+        function() {
+          const procedureModel =
+              new Blockly.procedures.ObservableProcedureModel(this.workspace);
+          procedureModel.insertParameter(
+              0,
+              new Blockly.procedures.ObservableParameterModel(
+                  this.workspace, 'test name'));
+
+          assertEventNotFired(
+            this.eventSpy,
+            Blockly.Events.ProcedureParameterCreate,
+            {},
+            this.workspace.id);
+        });
+
+    test(
+        'parameter delete events are fired when a parameter is deleted',
+        function() {
+          const procedureModel =
+              new Blockly.procedures.ObservableProcedureModel(this.workspace);
+          this.procedureMap.add(procedureModel);
+          const parameterModel =
+              new Blockly.procedures.ObservableParameterModel(
+                  this.workspace, 'test name');
+          procedureModel.insertParameter(0, parameterModel);
+          procedureModel.deleteParameter(0);
+
+          assertEventFired(
+            this.eventSpy,
+            Blockly.Events.ProcedureParameterDelete,
+            {
+              model: procedureModel,
+              parameter: parameterModel,
+              index: 0,
+            },
+            this.workspace.id);
+        });
+
+    test(
+        'parameter delete events are not fired if the procedure is ' +
+        'not in the map',
+        function() {
+          const procedureModel =
+              new Blockly.procedures.ObservableProcedureModel(this.workspace)
+                  .insertParameter(
+                      0,
+                      new Blockly.procedures.ObservableParameterModel(
+                          this.workspace, 'test name'));
+          procedureModel.deleteParameter(0);
+
+          assertEventNotFired(
+            this.eventSpy,
+            Blockly.Events.ProcedureParameterDelete,
+            {},
+            this.workspace.id);
+        });
+
+    test(
+        'parameter rename events are fired when a parameter is renamed',
+        function() {
+          const procedureModel =
+              new Blockly.procedures.ObservableProcedureModel(this.workspace);
+          this.procedureMap.add(procedureModel);
+          const parameterModel =
+              new Blockly.procedures.ObservableParameterModel(
+                  this.workspace, 'test name');
+          procedureModel.insertParameter(0, parameterModel);
+
+          parameterModel.setName('new name');
+
+          assertEventFired(
+            this.eventSpy,
+            Blockly.Events.ProcedureParameterRename,
+            {
+              model: procedureModel,
+              parameter: parameterModel,
+              oldName: 'test name',
```

Comment:
Does this event not include the new name?

Reply:
The new name exists on the parameter model, so there's no need to include it separately. At least as a user-visible property anyway. It's still included as a private property in the actual implementaiton so that undo and redo can be handled properly.

We can def make those proerties public if you want (esp cuz it's more consistent), I just didn't see a need to.